website/common/script/ops/buyArmoire.js

```diff
@@ -56,6 +58,16 @@ module.exports = function buyArmoire (user, req = {}, analytics) {
       user.flags.armoireEmpty = true;
     }
 
+    const officialPinnedItems = getOfficialPinnedItems(user);
+    if (isPinned(user, drop, officialPinnedItems)) {
+      const dropInfo = {
+        item: drop,
+        type: 'gear',
```

Comment:
with `gear`(gem purchasable) it doesn't seem to work for me, `marketGear` is the correct one (purchase by gold)